src/Storages/Statistics/StatisticsCountMinSketch.cpp

```diff
@@ -0,0 +1,111 @@
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/Statistics/StatisticsCountMinSketch.h>
+#include <Interpreters/convertFieldToType.h>
+#include <DataTypes/FieldToDataType.h>
+
+#if USE_DATASKETCHES
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+/// Constants chosen based on rolling dices, which provides an error tolerance of 0.1% (ε = 0.001) and a confidence level of 99.9% (δ = 0.001).
+/// And sketch the size is 152kb.
+static constexpr auto num_hashes = 7uz;
+static constexpr auto num_buckets = 2718uz;
+
+StatisticsCountMinSketch::StatisticsCountMinSketch(const SingleStatisticsDescription & stat_, DataTypePtr data_type_)
+    : IStatistics(stat_)
+    , sketch(num_hashes, num_buckets)
+    , data_type(data_type_)
+{
+}
+
+Float64 StatisticsCountMinSketch::estimateEqual(const Field & val) const
+{
+    if (data_type->isValueRepresentedByNumber())
+    {
+        /// 'val' maybe number or string, method 'convertFieldToType' will
+        ///     1. convert string to number, date, datetime, IPv4, Decimal etc
+        ///     2. return null if val larger than the range of data_type
+        auto val_converted = convertFieldToType(val, *data_type);
```

Comment:
What if it is an Int64 coverting to `data_type Int32` ? and the `data_type_converted` will also be Int32?

Reply:
For Int{8-64} are all represented by Filed(Int64), so when you are saying `it is an Int64`, I suppose that the `val` must be large than max_uint16. And `val_converted` will be null.